test/org/sagebionetworks/bridge/dynamodb/DynamoScheduledActivityDaoMockTest.java

```diff
@@ -67,55 +105,160 @@ public void before() {
+    public void getActivityHistoryV3() {
+        Activity activity = TestUtils.getActivity1();
+        String referentGuid = BridgeUtils.createReferentGuidIndex(activity, SCHEDULED_ON_START.toLocalDateTime());
+        String guid = activity.getGuid() + ":" + SCHEDULED_ON_START.toLocalDateTime();
+        
+        // Need less than an entire page of records
+        List<Item> items = new ArrayList<>();
+        for (int i=0; i < 10; i++) {
+            Item item = new Item().withString("guid", guid + ((i == 0) ? "" : i))
+                    .with("referentGuid", referentGuid).with("healthCode", HEALTH_CODE);
+            items.add(item);
         }
+        when(outcome.getItems()).thenReturn(items);
+        when(indexHelper.query(any())).thenReturn(outcome);
         
-        QueryResultPage<DynamoScheduledActivity> queryResultPage = (QueryResultPage<DynamoScheduledActivity>)mock(QueryResultPage.class);
-        when(queryResultPage.getResults()).thenReturn(dynamoActivities);
-        when(queryResultPage.getLastEvaluatedKey()).thenReturn(null);
+        List<Object> activities = new ArrayList<>();
+        for (int i=0; i < 10; i++) {
+            DynamoScheduledActivity oneActivity = new DynamoScheduledActivity(); 
+            // These values are mostly to get us through sorting, and are not that important
+            oneActivity.setGuid(guid + ((i == 0) ? "" : i));
+            oneActivity.setReferentGuid(referentGuid);
+            activities.add(oneActivity);
+        }
+        Map<String, List<Object>> resultMap = ImmutableMap.of("guid", activities);
+        when(mapper.batchLoad(any(List.class))).thenReturn(resultMap);
         
-        when(mapper.queryPage((Class<DynamoScheduledActivity>) any(Class.class),
-            (DynamoDBQueryExpression<DynamoScheduledActivity>) any(DynamoDBQueryExpression.class)))
-            .thenReturn(queryResultPage);
+        ForwardCursorPagedResourceList<ScheduledActivity> results = activityDao.getActivityHistoryV3(HEALTH_CODE,
+                ActivityType.TASK, activity.getGuid(), SCHEDULED_ON_START, SCHEDULED_ON_END,
+                null, PAGE_SIZE);
         
-        // Mock a batch load of the activities
-        Map<String,List<Object>> results = Maps.newHashMap();
-        results.put("some-table-name", new ArrayList<Object>(activities));
-        when(mapper.batchLoad(any(List.class))).thenReturn(results);
+        assertEquals(10, results.getItems().size());
+        assertNull(results.getNextPageOffsetKey());
+        assertNull(results.getRequestParams().get("offsetKey"));
+        assertEquals(PAGE_SIZE, (int)results.getRequestParams().get("pageSize"));
+        assertEquals(SCHEDULED_ON_START.toString(), results.getRequestParams().get("scheduledOnStart"));
+        assertEquals(SCHEDULED_ON_END.toString(), results.getRequestParams().get("scheduledOnEnd"));
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void getActivityHistoryV3NoOffsetKey() {
```

Comment:
Minor: I think this test and the previous test got their names swapped.

Reply:
swapped back